pandas/tests/test_index.py

```diff
@@ -2447,6 +2443,567 @@ def test_slice_keep_name(self):
         idx = Int64Index([1, 2], name='asdf')
         self.assertEqual(idx.name, idx[1:].name)
 
+class TestRangeIndex(Numeric, tm.TestCase):
+    _holder = RangeIndex
+    ### what does the following do?
+    #_multiprocess_can_split_ = True
+
+    def setUp(self):
+        self.indices = dict(index = RangeIndex(0, 20, 2))
+        self.setup_indices()
+
+    def create_index(self):
+        return RangeIndex(5)
+
+    def test_too_many_names(self):
+        def testit():
+            self.index.names = ["roger", "harold"]
+        assertRaisesRegexp(ValueError, "^Length", testit)
+
+    def test_constructor(self):
+        index = RangeIndex(5)
+        expected = np.arange(5, dtype=np.int64)
+        self.assert_numpy_array_equal(index, expected)
```

Comment:
its that you *need* a dtype. But you need to have the same signature as an ``Int64Index``. because that is how ``Index`` passes things thru. You can raise if its not None and not 'int64', its really for compat only.

The thing is since you can get many types of inputs (e.g. you could get a list/ndarray that is not-convertible to a ``RangeIndex``; should test this and I think raise), but you could also get ``np.arange(...)`` which works (I think I saw some inferernce logic somewhere for this, e.g. given an array determine if its convertible to a RI?)